src/app/Models/Traits/HasIdentifiableAttribute.php

```diff
@@ -23,9 +23,9 @@ public function identifiableAttribute()
         // in model we use this try/catch block. If the property exists we return it, if not (exception raised)
         // we return the result of our column name guessing.
         try {
-            return ! is_array($this->identifiableAttribute) ? [$this->identifiableAttribute] : $this->identifiableAttribute;
+            return $this->identifiableAttribute;
         } catch (Exception $e) {
-            return [static::guessIdentifiableColumnName()];
+            return static::guessIdentifiableColumnName();
         }
```

Comment:
If Laravel does not get mad with you it should work. Thing is, if you did not defined the property identifiableAttribute when returning $this->identifiableAttribute,  it would run $this->identifiableAttribute() and expect that it would return a relation instance. 

Checking if property_exists should avoid the error as we would only call it if the property exists. It's the same working principle but easier to read/understand and does not expect an exception to be thrown to proceed with code.